### PR TITLE
Jb/enable org browser for scratch orgs

### PR DESCRIPTION
**Pull request by jeffb-sfdc**
### What does this PR do?
This PR adds the Org Browser feature to scratch orgs.  It also enables the deploy & retrieve commands for scratch orgs as well.

### What issues does this PR fix or reference?
#@W-10931259@, @W-10931262@, @W-10931278@

### Functionality Before
The Org Browser, along with the Deploy and Retrieve commands were not available when the org was a scratch org.

### Functionality After
The Org Browser, along with the Deploy and Retrieve commands are now available when the org is a scratch org.

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -121,14 +121,14 @@
         {
           "id": "metadata",
           "name": "Metadata",
-          "when": "sfdx:project_opened && sfdx:default_username_has_no_change_tracking"
+          "when": "sfdx:project_opened && sfdx:enable_org_browser"
```
Comment (jeffb-sfdc) [inline packages/salesforcedx-vscode-core/package.json:124]: When I first started working on this, it took me a while to figure out that "sfdx:default_username_has_no_change_tracking" meant it was for a developer edition org, and "sfdx:default_username_has_change_tracking" meant it was a scratch org.  I renamed these functions (used for the where clauses) to what they enable/disable (sfdx:enable_deploy_and_retrieve_commands and sfdx:enable_push_and_pull_commands), and also added sfdx:enable_org_browser)

File: packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts [hunk 2]
```diff
@@ -17,24 +17,36 @@ export enum OrgType {
 export async function getWorkspaceOrgType(
   defaultUsernameOrAlias?: string
 ): Promise<OrgType> {
-  if (isNullOrUndefined(defaultUsernameOrAlias)) {
+  if (defaultUsernameOrAlias === null || defaultUsernameOrAlias === undefined) {
```
Comment (jeffb-sfdc) [inline packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts:20]: I changed this b/c I was getting a warning that isNullOrUndefined() is deprecated

File: packages/salesforcedx-vscode-core/test/vscode-integration/context/workspaceOrgType.test.ts [hunk 3]
```diff
@@ -136,8 +137,117 @@ describe('getWorkspaceOrgType', () => {
   });
 });
 
+describe.only('setWorkspaceOrgTypeWithOrgType', () => {
```
Comment (jeffb-sfdc) [inline packages/salesforcedx-vscode-core/test/vscode-integration/context/workspaceOrgType.test.ts:140]: These tests don't fully pass anymore.  They were written before I renamed enableOrgBrowser() to setEnableOrgBrowser(), and added enableOrgBrowser().  They just need a little cleanup and I will be working on them later today.

**Comment by gbockus-sf** [comment]
Verified this PR is linked in the associated WI.  Will need to revisit code with all the shifting around due to core v3.  Closing for now to be picked up when @W-10931259@ is picked up.

**Pull request closed without merging**